components/tools/OmeroM/src/annotations/unlinkAnnotations.m

```diff
@@ -0,0 +1,161 @@
+%
+% INPUT ARGUMENTS
+% session     omero.api.ServiceFactoryPrxHelper object
+%
+%               client = loadOmero('demo.openmicroscopy.org', 4064)
+%               session = client.createSession(username, password)
+%
+% parentType  'project' | 'dataset' | 'image' | 'screen' | 'plate' | 
+%             'plateacquisition' | 'roi'
+%
+% parentID    vector of non-negative integers
+%             parentID can be found via OMERO GUI
+%
+%
+% annt        omero.model.Annotation object or its array | [] (default)
+%
+%             (Optional) omero.model.Annotation object (supports
+%             omero.model.MapAnnotationI, omero.model.TagAnnotationI,
+%             omero.model.CommentAnnotationI, and
+%             omero.model.FileAnnotationI at the moment) to be unlinked
+%             from parent If empty or not specified, all linked
+%             MapAnnotation objects will be unlinked.
+%            
+%             For example, to obtain MapAnnotationI object(s) from an image:
+%
+%               annt = getObjectAnnotations(session, 'map', 'image', parentID);
+%
+%             or
+%
+%               annt = getImageTagAnnotations(session, ids)
+% 
+%             annt can be empty, scalar or a vector.
+%
+%             Also you can use following functions or similar.
+%
+%               annt = getAnnotations(session,ids,'tags')
+%               annt = getTagAnnotations(session,tagID)
+%
+%
+%
+% Written by Kouichi C. Nakamura Ph.D.
+% MRC Brain Network Dynamics Unit
+% University of Oxford
+% kouichi.c.nakamura@gmail.com
+% 09-Jun-2018 03:27:05
+%
+% See also
+% linkAnnotation
+
+objects = getObjectTypes();
+
+p = inputParser;
+p.addRequired('session',@(x) isjava(x));
+p.addRequired('parentType',@(x) ischar(x) && ismember(x, {objects.name}));
+p.addRequired('parentID',@(x) isnumeric(x) && isvector(x));
+
+p.addOptional('annt',[],@(x) isempty(x) || all(isa(x, 'omero.model.Annotation')));
+p.parse(session,parentType,parentID,varargin{:});
+
+annt = p.Results.annt;
+
+
+% https://docs.openmicroscopy.org/omero/5.4.6/developers/Model/StructuredAnnotations.html
+% Project
+% Dataset
+% Pixels
+% OriginalFile
+% PlaneInfo
+% Roi
+% Channel
+% Folder
+
+% https://github.com/openmicroscopy/openmicroscopy/blob/develop/examples/Training/matlab/ReadData.m#L334
+
+
+switch parentType
+   
+    case 'project'
+        ParentType = 'Project';
+    case 'dataset'
+        ParentType = 'Dataset';
+    case 'image'
+        ParentType = 'Image';
+    case 'screen'
+        ParentType = 'Screen';
+    case 'plate'
+        ParentType = 'Plate';
+    case 'plateacquisition'
+        ParentType = 'PlateAcquisition';
+    case 'roi'
+        ParentType = 'Roi';
+end
+
+parents = session.getQueryService().findAllByQuery(...
+    [sprintf('select obj from %s as obj left outer join fetch obj.annotationLinks as link join fetch link.child as annotation where obj.id =  ',ParentType), ...
+    num2str(parentID)], []);
+
+% images = session.getQueryService().findAllByQuery(...
```

Comment:
I would simply delete these lines rather than having it commented without an explanation